src/endpoints/authentication/cose_auth.cpp

```diff
@@ -96,26 +102,40 @@ namespace ccf
       header_items[GOV_MSG_TYPE].uLabelType = QCBOR_TYPE_TEXT_STRING;
       header_items[GOV_MSG_TYPE].uDataType = QCBOR_TYPE_TEXT_STRING;
 
-      auto gov_msg_proposal_id = COSE_HEADER_PARAM_PROPOSAL_ID;
+      auto gov_msg_proposal_id = COSE_HEADER_PARAM_MSG_PROPOSAL_ID;
       header_items[GOV_MSG_PROPOSAL_ID].label.string =
         UsefulBuf_FromSZ(gov_msg_proposal_id);
       header_items[GOV_MSG_PROPOSAL_ID].uLabelType = QCBOR_TYPE_TEXT_STRING;
       header_items[GOV_MSG_PROPOSAL_ID].uDataType = QCBOR_TYPE_TEXT_STRING;
 
+      auto gov_msg_proposal_created_at = COSE_HEADER_PARAM_MSG_CREATED_AT;
+      header_items[GOV_MSG_CREATED_AT].label.string =
+        UsefulBuf_FromSZ(gov_msg_proposal_created_at);
+      header_items[GOV_MSG_CREATED_AT].uLabelType = QCBOR_TYPE_TEXT_STRING;
+      // Although this is really uint, specify QCBOR_TYPE_INT64
+      // QCBOR_TYPE_UINT64 only matches uint values that are greater than
+      // INT64_MAX
+      header_items[GOV_MSG_CREATED_AT].uDataType = QCBOR_TYPE_INT64;
```

Comment:
Question about QCBOR, not this PR: This seems mad? We're telling the decoder what type to expect, but need to know at this point whether we expect the _value_ to be greater than `INT64_MAX`, to pass a different value? Not a problem in this instance, for a few aeons.

Reply:
It surprised me too, but comments suggest it is by design:

https://github.com/laurencelundblade/QCBOR/blob/37286c08a00d7516cf2da173a48d1bfdd5260eff/inc/qcbor/qcbor_decode.h#L221
https://github.com/laurencelundblade/QCBOR/blob/4e808ba6452134c04247e956f2a81abef2442a6c/src/qcbor_decode.c#L765

If the value is greater, I expect the decoding will fail with a type mismatch.